llvm/lib/Transforms/Scalar/CorrelatedValuePropagation.cpp

```diff
@@ -528,17 +530,35 @@ static bool processAbsIntrinsic(IntrinsicInst *II, LazyValueInfo *LVI) {
 }
 
 // See if this min/max intrinsic always picks it's one specific operand.
+// If not, check whether we can canonicalize signed minmax into unsigned version
 static bool processMinMaxIntrinsic(MinMaxIntrinsic *MM, LazyValueInfo *LVI) {
   CmpInst::Predicate Pred = CmpInst::getNonStrictPredicate(MM->getPredicate());
   LazyValueInfo::Tristate Result = LVI->getPredicateAt(
       Pred, MM->getLHS(), MM->getRHS(), MM, /*UseBlockValue=*/true);
```

Comment:
Can you please also replace this getPredicateAt() use with comparison on getConstantRangeAtUse()? It makes very little sense to do both. (getPredicateAt can be better in rare cases involving equality predicates, but it shouldn't be relevant here.)